packages/network/lib/cors.ts

```diff
@@ -71,26 +76,118 @@ export function getDomainNameFromParsedHost (parsedHost: ParsedHost) {
   return _.compact([parsedHost.domain, parsedHost.tld]).join('.')
 }
 
-export function urlMatchesOriginPolicyProps (urlStr, props) {
-  // take a shortcut here in the case
-  // where remoteHostAndPort is null
-  if (!props) {
+/**
+ * same-origin: Whether or not a a urls scheme, port, and host match. @see https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy
+ * same-super-domain-origin: Whether or not a url's scheme, domain, top-level domain, and port match
+ * same-site: Whether or not a url's scheme, domain, and top-level domain match to determine whether or not
+ * a cookie is considered first-party. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#third-party_cookies
```

Comment:
Feels like the cookie portion should be moved to where this function is called and not part of the generic function's documentation.

Reply:
fixed in [10a64c9](https://github.com/cypress-io/cypress/pull/23879/commits/10a64c9e2e76a972eba25e171ff414e4f202b274)

Reply:
From your linked definition:

> In some contexts, the scheme is also considered when differentiating sites. This would make http://vpl.ca and https://vpl.ca different sites. Including the scheme prevents an insecure (HTTP) site from being treated as the same site as a secure (HTTPS) site. A definition that considers the scheme is sometimes called a schemeful same-site. This stricter definition is applied in the rules for handling [SameSite](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite) cookies.

It seems like we should name it `schemeful-same-site` in case we need to add the looser same-site in the future.

Reply:
should we go with `schemeful-same-site`  but keep the samesite naming nomenclature for the cors functions?

Reply:
updated the policy in b8f8b6ab85